runtime/doc/quickfix.txt

```diff
@@ -1278,6 +1278,31 @@ For writing a compiler plugin, see |write-compiler-plugin|.
 
 Use the |compiler-make| plugin to undo the effect of a compiler plugin.
 
+CPPCHECK			*quickfix-cppcheck* *compiler-cppcheck*
+
+Use g/b:c_cppcheck_params to set cppcheck parameters. The global
+settings by default include
+
+- `--verbose`: Enables verbose output.
+- `--force`: Forces checking of all configurations.
+- `--inline-suppr`: Allows inline suppressions.
+- `--enable=...`: Enables specific checks like warnings, style, performance,
+  portability, information, and missing includes.
+- `-j`: Utilizes multiple processors if available, determined by the
+  `getconf` command if available (requires omitting the unusedFunction check)
+
+For C++ files (`filetype == 'cpp'`), the `--language=c++` option is added to
+ensure Cppcheck treats the file as C++.
+
+By default the directories in &path are passed as include directories.
+These can be set by g/b:c_cppcheck_includes as a list of '-I' flags.
```

Comment:
Finding includes this way is almost unusable for large projects with many include paths.

It would be better to support include paths from `compile_commands.json` (already used by many other tools such as clangd, clang-tidy, include-what-you-use).  I see that cppcheck can use this too with `cppcheck --project=compile_commands.json`